security/pfSense-pkg-zeek/files/usr/local/pkg/zeek.inc

```diff
@@ -0,0 +1,982 @@
+			//stop_service("zeek");
+			mwexec(ZEEK_ETC . "/rc.d/zeek.sh stop");
+		}
+	}}
+
+/*
+ * rc scripts, services and cronjobs
+ */
+
+ // Handle cronjob install/uninstall
+ function zeek_install_cron($should_install) {
+	 if (platform_booting()) {
+		 return;
+	 }
+	 $cron_cmd = ZEEK_LOCALBASE . "/bin/zeekctl cron ";
+
+	 if (($should_install) && (zeek_enabled())) {
+		 log_error("[zeek] Adding cronjobs ...");
+		 install_cron_job("{$cron_cmd}", $should_install, "*/5", "*", "*", "*", "*", "root");
+	 } else {
+		 log_error("[zeek] Removing cronjobs ...");
+		 install_cron_job("{$cron_cmd}", false);
+	 }
+ }
+
+ function zeek_write_rcfile(){
+	 $zeek_etc_dir = ZEEK_ETC;
+	 $rcfile = array();
+	 $rcfile['file'] = 'zeek.sh';
+	 $rcfile['start'] = <<< EOD
+{$zeek_etc_dir}/rc.d/zeek onestart
+
+EOD;
+
+	 $rcfile['stop'] = <<< EOD
+{$zeek_etc_dir}/rc.d/zeek stop
+# Just to be sure...
+sleep 5
+if [ -n "`/bin/ps auxw | /usr/bin/grep "zeek" | /usr/bin/awk '{print $2}'`" ]; then
+/usr/bin/killall -9 zeek 2>/dev/null
+fi
+EOD;
+
+	 write_rcfile($rcfile);
+ }
+
+ /*
+ * Zeek package install/uninstall
+ */
+
+ function zeek_install_command() {
+	 global $g, $config;
+
+	 /* Hard kill any running zeek processes that may have been started by any
+	 of the pfSense scripts such as check_reload_status() or rc.start_packages*/
+	 if(is_process_running("zeek")) {
+		 exec("/usr/bin/killall -z zeek");
+		 sleep(2);
+	 }
+	 // Turn off tcp segmentation offload.
+	 exec("sysctl net.inet.tcp.tso=0");
+
+	 // creating a backup file of the original zeekctl.cfg no matter if user checked this or not
+	 if (!file_exists(ZEEK_ETC . "/zeekctl.cfg.backup")) {
+		 log_error("[zeek] Creating backup of the original file to " . ZEEK_ETC . "/zeekctl.cfg.backup");
+		 copy(ZEEK_ETC . "/zeekctl.cfg", ZEEK_ETC . "/zeekctl.cfg.backup");
+	 }
+
+	 // creating a backup file of the original /etc/node no matter if user checked this or not
+	 if (!file_exists(ZEEK_ETC . "/node.cfg.backup")) {
+		 log_error("[zeek] Creating backup of the original file to " . ZEEK_ETC . "/node.cfg.backup");
+		 copy(ZEEK_ETC . "/node.cfg", ZEEK_ETC . "/node.cfg.backup");
+	 }
+
+	 // creating a backup file of the original /etc/networks no matter if user checked this or not
+	 if (!file_exists(ZEEK_ETC . "/networks.cfg.backup")) {
+		 log_error("[zeek] Creating backup of the original file to " . ZEEK_ETC . "/networks.cfg.backup");
+		 copy(ZEEK_ETC . "/networks.cfg", ZEEK_ETC . "/networks.cfg.backup");
+	 }
+	 // Change owner of zeek created files
+	 exec("chown -R root:wheel /usr/local/logs");
+
+	 exec("chown -R root:wheel /usr/local/spool");
+
+	 unlink_if_exists("/usr/local/etc/rc.d/zeek.sh");
+
+	 // create zeek rcfile
+	 zeek_write_rcfile();
+
+	 write_config("[zeek] Package installed.");
+ }
+
+
+ function zeek_deinstall_command() {
+	 zeek_install_cron(false);
+	 mwexec("/bin/ps awux | /usr/bin/egrep -i 'zeek -f|\(zeek\)' | /usr/bin/awk '{ print $2 }' | /usr/bin/xargs kill");
```

Comment:
This seems overly complicated, why not just `/usr/bin/killall zeek` unconditionally?